Octokit.Tests.Integration/Helpers/ApplicationTestAttribute.cs

```diff
@@ -8,16 +9,23 @@ namespace Octokit.Tests.Integration
 {
     public class ApplicationTestDiscoverer : IXunitTestCaseDiscoverer
     {
-        public IEnumerable<IXunitTestCase> Discover(ITestMethod testMethod, IAttributeInfo factAttribute)
+        readonly IMessageSink diagnosticMessageSink;
+
+        public ApplicationTestDiscoverer(IMessageSink diagnosticMessageSink)
+        {
+            this.diagnosticMessageSink = diagnosticMessageSink;
+        }
+
+
+        public IEnumerable<IXunitTestCase> Discover(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
         {
             if (String.IsNullOrWhiteSpace(Helper.ClientId)
                 && String.IsNullOrWhiteSpace(Helper.ClientSecret))
             {
-                yield return new SkipTestCase(testMethod,
-                    "Environment variables are not set for this test - set OCTOKIT_CLIENTID and OCTOKIT_CLIENTSECRET");
+                return Enumerable.Empty<IXunitTestCase>();
             }
 
-            yield return new XunitTestCase(testMethod);
+            return new[] { new XunitTestCase(diagnosticMessageSink, discoveryOptions.MethodDisplayOrDefault(), testMethod) };
```

Comment:
The biggest change from where we were is the `XunitTestCase` ctors - I stole this code from the https://github.com/xunit/samples.xunit repo.